lab3/client/src/views/generator.js

```diff
@@ -0,0 +1,58 @@
+export default function (store) {
+	return class GeneratorComponent extends window.HTMLElement {
+		constructor () {
+			super();
+			this.store = store;
+			this.connectedCallBack();
+			this.onStateChange = this.handleStateChange.bind(this);
+		}
+
+		handleStateChange (newState) {
+			this.innerHTML = `<div>
+				<div class="center">${newState.generators[this.dataset.id].name}</div>
+				<div class="center">Quantity: ${newState.generators[this.dataset.id].quantity}</div>
+
+				<p class="center">${newState.generators[this.dataset.id].description}</p>
+				<p class="center">Rate: ${newState.generators[this.dataset.id].rate}</p>
+				<button class="center">Buy ${newState.generators[this.dataset.id].name}</button>
```

Comment:
Should display the cost in the button using `getCost` function or user has no idea how much a generator costs.